timesketch/lib/analyzers/feature_plugins/winevt_features.py

```diff
@@ -0,0 +1,204 @@
+    """A plugin for Windows event log feature extraction."""
+
+    NAME = "winevt_feature_extraction"
+    DISPLAY_NAME = "Windows Event Log Feature Extraction"
+    DESCRIPTION = (
+        "This plugin extracts Windows event logs attributes from plaso output"
+        " attribute `strings`"
+    )
+
+    EVENT_FIELDS = ["strings"]
+
+    def run_plugin(self, name: str, config: dict) -> str:
+        """Extracts features from events.
+
+        Args:
+            name (str): Feature extraction name.
+            config (dict): A dict that contains the configuration for the feature
+                extraction.
+
+        Returns:
+            str: Returns summary of the feature extraction.
+        """
+        return self.extract_features(name, config)
+
+    def validate_feature_config(self, name: str, config: dict) -> None:
+        """Validates the name and configuration.
+
+        Args:
+            name (str): Name of the feature.
+            config (dict): Configuration parameter.
+
+        Raises:
+            ValueError: Raises ValueError for value type errors.
+        """
+        if not name:
+            raise ValueError("Feature name is empty")
+
+        if not config:
+            raise ValueError("Feature configuration value is empty")
+
+        if not (
+            isinstance(config.get("source_name"), list)
+            or isinstance(config.get("provider_identifier"), list)
+        ):
+            raise ValueError("Either source_name or provider_identifier must be a list")
+
+        if not isinstance(config.get("event_identifier"), int):
+            raise ValueError("event_identifier is missing or has invalid value")
+
+        if not isinstance(config.get("event_version"), int):
+            raise ValueError("event_version is missing or has invalid value")
+
+    def extract_features(self, name: str, config: dict) -> str:
+        """Extracts features from events.
+
+        Args:
+            name (str): Features extraction name.
+            config (dict): A dict that contains the configuration fo the
+                feature extraction.
+
+        Returns:
+            str: Returns summary of the feature extraction for `feature_name`.
+        """
+        self.validate_feature_config(name, config)
+
+        source_name = config.get("source_name", [""])[0]
+        provider_identifier = config.get("provider_identifier", [""])[0]
+
+        # If event_identifier and event_version does not exist or invalid value type,
+        # it will throw exception.
+        #
+        # Note: Each feature extraction is run as a new celery task so it only affects
+        # specific feature extraction.
+        event_identifier = int(config.get("event_identifier"))
+        event_version = int(config.get("event_version"))
+
+        mappings = config.get("mapping")
+
+        if not mappings:
+            raise ValueError(f"mapping value for {name} is empty")
```

Comment:
This is not needed anymore since it is covered by the `validate_feature_config` function.

```suggestion
```